WSLBlobNFS/wsl2_linux_script.sh

```diff
@@ -60,54 +166,184 @@ function onetime_samba_setup ()
+        mountPath=${nametokens[-1]}
+
+        # check if the last element is a path
+        if [[ ${mountPath:0:1} == "/" ]]; then
+            # check if the path exists
+            if [[ ! -d $mountPath ]]; then
+                mkdir -p $mountPath
+                vecho "Created $mountPath"
+            else
+                vecho "Mount point $mountPath already exists"
+            fi
+        else
+            eecho "Mount point $mountPath is not a path"
+            exit 1
+        fi
+
+        mpath=${mountPath:1}
+
+        # replace all / with -
+        shareName=${mpath//\//-}
+    else
+        # RemoteHost: <account-name>.blob.core.windows.net:/<account-name>/<container-name>
+
+        # A random non existent mount path to mount the NFS share
+        randomnumber=$RANDOM
+        mountPath="/mnt/nfsv3share-$randomnumber"
+
+        while [[ -e $mountPath ]]; do
+            randomnumber=$RANDOM
+            mountPath="/mnt/nfsv3share-$randomnumber"
+        done
+
+        mkdir -p $mountPath
+        vecho "Created $mountPath"
+
+        mountCommand="mount -t nfs -o vers=3,proto=tcp $mountparameter $mountPath"
+        shareName="nfsv3share-$randomnumber"
+    fi
+
+    # Check if the mount path is already mounted
+    mount | grep -q $mountPath
+    if [[ $? == 0 ]]; then
+        eecho "Mount path $mountPath is already mounted. Use a different mount path."
+        exit 1
+    fi
+
+    vecho "Mounting NFS share.."
+
+    # quote the mount command to preserve the spaces
+    mount_nfs "$mountCommand" "$mountPath"
+
+    if [[ $? != 0 ]]; then
+        eecho "Failed to mount NFS share with: $mountCommand"
+        exit 1
+    fi
+
+    vecho "Done NFS mounting."
+
+    vecho "Exporting NFS share via Samba.."
+    share_via_samba "$shareName" "$mountPath"
+
+    if [[ $? != 0 ]]; then
+        eecho "Failed to export NFS share via Samba with: $shareName, $mountPath"
+
+        # Unmount the NFS share
+        unmount_nfs $mountPath
+        exit 1
+    fi
+
+    echo "$shareName" > $tempFilePath
+    vecho "Saved the share name ($sharename) to $tempFilePath"
+}
+
+# Unmount NFS
+# $1: mount path
+function unmount_nfs ()
+{
+    if [[ $# != 1 ]]; then
+        eecho "Usage: unmount_nfs <mount path>"
+        exit 1
+    fi
+
+    mntpath=$1
+    umount $mntpath
+
+    # If the umount command fails with 32, then the NFS share is already unmounted
+    if [[ $? != 0 && $? != 32 ]]; then
+        eecho "Failed to unmount NFS share at: $mntpath"
+        return 1
+    fi
+
+    vecho "Unmounted NFS mount at: $mntpath"
+    return 0
 }
 
 # Unmount smb share and NFS share
-# $1: export name
+# $1: share name
 # To-do:
 # 1. Take a lock on the smb file
 function unmount_share ()
 {
+    smbsharename=$1
     mntpath=""
     foundshare=false
     startline=0
     endline=0
     linenum=0
 
+    # Take a backup of smb.conf to restore it if the script fails
+    cp /etc/samba/smb.conf /etc/samba/smb.conf.bak
```

Comment:
in case .bak file already exists and cp is aliased to be interactive, do the following to avoid a prompt here
rm -f /etc/samba/smb.conf.bak
cp -f /etc/samba/smb.conf /etc/samba/smb.conf.bak

-f to cp is optional but it's a good practice